File: src/p0007/java/solution.java
class Solution {
    public int reverse(int x) {
        int y = 0;
        for (; x != 0; x /= 10) {
            if (x > 0 && (y > Integer.MAX_VALUE / 10 || (y == Integer.MAX_VALUE / 10 && x % 10 > Integer.MAX_VALUE % 10))) {
                y = 0;
                break;
            }
            if (x < 0 && (y < Integer.MIN_VALUE / 10 || (y == Integer.MIN_VALUE / 10 && x % 10 < Integer.MIN_VALUE % 10))) {
                y = 0;
                break;
            }
            y = y * 10 + x % 10;
        }
        return y;
    }
}

File: src/p0007/rust/solution.rs
use std::i32::MAX;
use std::i32::MIN;

impl Solution {
    pub fn reverse(x: i32) -> i32 {
        let (mut x, mut y) = (x, 0);
        while x != 0 {
            if x > 0 && (y > MAX / 10 || (y == MAX / 10 && x % 10 > MAX % 10)) {
                y = 0;
                break;
            }
            if x < 0 && (y < MIN / 10 || (y == MIN / 10 && x % 10 < MIN % 10)) {
                y = 0;
                break;
            }
            y = y * 10 + x % 10;
            x /= 10;
        }
        return y;
    }
}
